alacritty/src/display.rs

```diff
@@ -0,0 +1,461 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! The display subsystem including window management, font rasterization, and
+//! GPU drawing.
+use std::f64;
+use std::fmt;
+use std::time::Instant;
+
+use glutin::dpi::{PhysicalPosition, PhysicalSize};
+use glutin::event_loop::EventLoop;
+use log::{debug, info};
+use parking_lot::MutexGuard;
+
+use font::{self, Rasterize, Size};
+
+use alacritty_terminal::config::StartupMode;
+use alacritty_terminal::event::{Event, OnResize};
+use alacritty_terminal::index::Line;
+use alacritty_terminal::message_bar::MessageBuffer;
+use alacritty_terminal::meter::Meter;
+use alacritty_terminal::renderer::rects::{RenderLines, RenderRect};
+use alacritty_terminal::renderer::{self, GlyphCache, QuadRenderer};
+use alacritty_terminal::term::color::Rgb;
+use alacritty_terminal::term::{RenderableCell, SizeInfo, Term};
+
+use crate::config::Config;
+use crate::event::Resize;
+use crate::window::{self, Window};
+
+#[derive(Debug)]
+pub enum Error {
+    /// Error with window management
+    Window(window::Error),
+
+    /// Error dealing with fonts
+    Font(font::Error),
+
+    /// Error in renderer
+    Render(renderer::Error),
+
+    /// Error during buffer swap
+    ContextError(glutin::ContextError),
+}
+
+impl std::error::Error for Error {
+    fn cause(&self) -> Option<&dyn (std::error::Error)> {
+        match *self {
+            Error::Window(ref err) => Some(err),
+            Error::Font(ref err) => Some(err),
+            Error::Render(ref err) => Some(err),
+            Error::ContextError(ref err) => Some(err),
+        }
+    }
+
+    fn description(&self) -> &str {
+        match *self {
+            Error::Window(ref err) => err.description(),
+            Error::Font(ref err) => err.description(),
+            Error::Render(ref err) => err.description(),
+            Error::ContextError(ref err) => err.description(),
+        }
+    }
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match *self {
+            Error::Window(ref err) => err.fmt(f),
+            Error::Font(ref err) => err.fmt(f),
+            Error::Render(ref err) => err.fmt(f),
+            Error::ContextError(ref err) => err.fmt(f),
+        }
+    }
+}
+
+impl From<window::Error> for Error {
+    fn from(val: window::Error) -> Error {
+        Error::Window(val)
+    }
+}
+
+impl From<font::Error> for Error {
+    fn from(val: font::Error) -> Error {
+        Error::Font(val)
+    }
+}
+
+impl From<renderer::Error> for Error {
+    fn from(val: renderer::Error) -> Error {
+        Error::Render(val)
+    }
+}
+
+impl From<glutin::ContextError> for Error {
+    fn from(val: glutin::ContextError) -> Error {
+        Error::ContextError(val)
+    }
+}
+
+/// The display wraps a window, font rasterizer, and GPU renderer
```

Comment:
A comma before `GPU` seems odd

Reply:
FYI: [15 Reasons You Should Definitely Use The Oxford Comma](https://www.buzzfeed.com/sarahhenrich/15-reasons-why-you-should-use-the-oxford-comma-yr75)

Reply:
Thanks for the link, I just automatically mapped Russian punctuation on the English one here for some reason (they are very similar most of the time though). 

I just can't read the second example the way it's described in your link, since this thing doesn't even exist in Russian ( we are using `:` for such things ).

Glad to learn something new.